### DISCUSSION TITLE
vercel alias using --token for authentication fails

### DISCUSSION THREAD

**Discussion by random-tech** [vercel/community#1781] [Help] [answered]
### Summary

As part of our deployment pipeline we would like to assign the preview-deployments to a stable url. I've successfully done that locally like this:
```
DEPLOY_URL=$(vercel deploy --prebuilt)
vercel alias set $DEPLOY_URL <static url>
```
But when running this from our CI/CD pipeline we need to use tokens to authenticate, which fails:
```
DEPLOY_URL=$(vercel deploy --prebuilt --token=$VERCEL_TOKEN)
vercel alias set $DEPLOY_URL <static url> --token=$VERCEL_TOKEN
```
giving the error message "Error: deployment dpl_**** not found".
Running the same command locally using a token fails the same way, but running locally without a token works fine.

I've tried creating several different tokens with different scopes, in case that was the issue, but with no success.

Does anyone know how to get this to work?

**Comment by random-tech** [comment, marked as answer]
Turns out that for the `vercel alias` call for a team project you need to specify the team id using `--scope`.
So this worked for me: `vercel alias set $DEPLOY_URL <static-url> --token=$VERCEL_TOKEN --scope=$VERCEL_TEAM_ID`

Why you need to specify scope for the `alias` call, but not for the `deploy` call, is a bit of a mystery to me, but at least it now works.

**Comment by iamcrisb** [comment]
I swear this should be added to the docs, also it seems that vercel does not know the difference between id's and slugs.
